### PR TITLE
Suppress "pointer from int conversion" warnings on 32-bit builds

### PR DESCRIPTION
**Pull request by 1ace**
... and turn those into errors

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,6 +87,9 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
                         -Wextra
                         -Wno-unused-parameter
                         -Wno-missing-field-initializers
+                        -Werror=int-conversion
+                        -Werror=pointer-to-int-cast
+                        -Werror=int-to-pointer-cast
```
Comment (djdeath) [inline CMakeLists.txt]: If you're doing all of the other changes, do you still need those new flags?

Comment (1ace) [reply CMakeLists.txt]: The point is to avoid these happening again in the future, which will be checked on each PR when a CI is set up

Comment (djdeath) [reply CMakeLists.txt]: Oh apologies, I read it with `-Wno-error-*`

**Comment by djdeath** [comment]
Getting those warnings from g++ : 

```
cc1plus: warning: ‘-Werror=’ argument ‘-Werror=int-conversion’ is not valid for C++
cc1plus: warning: ‘-Werror=’ argument ‘-Werror=pointer-to-int-cast’ is not valid for C++
```

Might need some cmake foo to make it go away.

**Comment by djdeath** [comment]
Made it go away with : 

```
    add_compile_options(-Wall
                        -Wextra
                        -Wno-unused-parameter
                        -Wno-missing-field-initializers
                        -Werror=int-to-pointer-cast
                        -fno-strict-aliasing
                        -fno-builtin-memcmp
                        -fvisibility=hidden
                        "$<$<COMPILE_LANGUAGE:C>:-Werror=int-conversion>"
                        "$<$<COMPILE_LANGUAGE:C>:-Werror=pointer-to-int-cast>")
```

**Comment by 1ace** [comment]
I'll test this to make sure the flags aren't simply dropped and update the PR later :+1:

**Comment by djdeath** [comment]
Merged manually.

**Pull request closed without merging**